code/game/machinery/autolathe.dm

```diff
@@ -136,93 +358,131 @@
 				flick("autolathe_r",src)//plays glass insertion animation by default otherwise
 		use_power(min(1000, amount_inserted / 100))
 	updateUsrDialog()
+	//Begin processing to continue the queue if we had items in the queue
+	if(wants_operate)
+		begin_process()
 
-/obj/machinery/autolathe/Topic(href, href_list)
-	if(..())
+/obj/machinery/autolathe/proc/begin_process()
+	if(busy || operating || disabled)
+		return
+	var/requested_design_id = null
+	var/from_build_queue = FALSE
+	if(LAZYLEN(build_queue))
+		for(var/design_id in build_queue)
```

Comment:
Is this just a really bad way of getting the first item from a list

Reply:
Yes, a really really bad way

Reply:
build_queue[1] should get the first value for dictionary style lists I hope